jlm/llvm/opt/alias-analyses/PointerObjectSet.cpp

```diff
@@ -373,6 +390,44 @@ PointerObjectSet::Clone() const
   return std::make_unique<PointerObjectSet>(*this);
 }
 
+bool
+PointerObjectSet::HasIdenticalSolAs(const PointerObjectSet & other) const
+{
+  if (NumPointerObjects() != other.NumPointerObjects())
+    return false;
+
+  // Check that each pointer object has the same Sol set in both sets
+  for (PointerObjectIndex i = 0; i < NumPointerObjects(); i++)
+  {
+    if (HasEscaped(i) != other.HasEscaped(i))
+      return false;
+
+    if (IsPointingToExternal(i) != other.IsPointingToExternal(i))
+      return false;
+
+    auto & thisPointsToSet = GetPointsToSet(i);
+    auto & otherPointsToSet = other.GetPointsToSet(i);
+
+    for (auto thisPointee : thisPointsToSet.Items())
+    {
+      // Skip doubled-up pointers, as they do not affect the points-to set
+      if (HasEscaped(thisPointee) && IsPointingToExternal(i))
```

Comment:
Maybe introduce a method `IsDoubledUp()` that is used instead of this pattern (which is used at multiple places)?

Reply:
I rewrote it a bit and added a general `IsPointingTo`, that returns true for explicit, implicit, and doubled-up pointer/pointee relations.